### PR TITLE
Add ddtrace package

### PR DESCRIPTION
**Pull request by SarvarD**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/ddtrace/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/ddtrace-{{ version }}.tar.gz
+  sha256: dba66c7013a4352dfa20ed31e989001250056c89811d1cb5ee10e23dc98bac0a
+
+build:
+  number: 0
+  skip: true
```
Comment (scopatz) [inline recipes/ddtrace/meta.yaml:15]: This line prevents the package from building at all, and probably should be removed

Comment (SarvarD) [reply recipes/ddtrace/meta.yaml:15]: Thanks for the feedback! This one made me a little comfortable. This was autogenerated via `grayskull`, I'll update.

Comment (scopatz) [reply recipes/ddtrace/meta.yaml]: You should let grayskull know. That seems like a bug

File: recipes/ddtrace/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    - python
+    - tenacity >=5
+
+test:
+  imports:
+    - ddtrace
+    - ddtrace.bootstrap
+  commands:
+    - pip check
+    - ddtrace-run --help
```
Comment (mariusvniekerk) [inline recipes/ddtrace/meta.yaml:43]: doesn't look like you can just run these tests here.  I'd remove them for now

File: recipes/ddtrace/meta.yaml (hunk 1 above)
Comment (mariusvniekerk) [reply recipes/ddtrace/meta.yaml]: Looks like the selectors got removed when the comments were removed.  We should probably update the bot to state that selector lines are not comments?

File: recipes/ddtrace/meta.yaml (hunk 2 above)
Comment (SarvarD) [reply recipes/ddtrace/meta.yaml]: Thanks Marius, looks like you're right :)

File: recipes/ddtrace/meta.yaml (hunk 1 above)
Comment (SarvarD) [reply recipes/ddtrace/meta.yaml]: Right, this was me removing the "comment," which it seems is not just a cosmetic change. Maybe the information that's currently being captured via those comments can instead be part of the `yaml` longterm?

File: recipes/ddtrace/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - cython
+    - pip
+    - python
+    - setuptools_scm
+    - setuptools_scm >=4
```
Comment (scopatz) [inline recipes/ddtrace/meta.yaml:29]: You should only have one of the above two lines

Comment (SarvarD) [reply recipes/ddtrace/meta.yaml:28]: That makes sense, thank you

File: recipes/ddtrace/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/ddtrace-{{ version }}.tar.gz
+  sha256: dba66c7013a4352dfa20ed31e989001250056c89811d1cb5ee10e23dc98bac0a
+
+build:
+  number: 0
+  skip: true   # [py==30 or py==31 or py==32 or py==33 or py==34]
+  entry_points:
+    - ddtrace-run = ddtrace.commands.ddtrace_run:main
```
Comment (SarvarD) [inline recipes/ddtrace/meta.yaml:17]: It looks like these were specified just for the old tests:

```
  commands:
    - pip check
    - ddtrace-run --help
    - pyddprofile --help
```
Would it be reasonable to remove these?

Comment (scopatz) [reply recipes/ddtrace/meta.yaml:17]: You could probably remove the pip check

Comment (SarvarD) [reply recipes/ddtrace/meta.yaml:17]: I removed all of these in a previous diff - these were also generated by grayskull, but it looks like these commands need more complex environment setup to execute.

File: recipes/ddtrace/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    - ddtrace
+    - ddtrace.bootstrap
+  requires:
+    - pip
+
+about:
+  home: https://github.com/DataDog/dd-trace-py
+  summary: Datadog tracing code
+  license: BSD-3-Clause
+  license_file: LICENSE
```
Comment (scopatz) [inline recipes/ddtrace/meta.yaml:48]: Please include the actual BSD license too

Comment (SarvarD) [reply recipes/ddtrace/meta.yaml:48]: Just added it - thank you for catching that

**Review by mariusvniekerk — APPROVED** [review]

**Pull request merged**